bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -0,0 +1,687 @@
+                        resource.getColorTemperaturePercentState(mirekSchemaFrom(resource)), fullUpdate);
+                updateState(HueBindingConstants.CHANNEL_2_COLOR_TEMPERATURE_ABS,
+                        resource.getColorTemperatureKelvinState(), fullUpdate);
+                updateState(HueBindingConstants.CHANNEL_COLOR, resource.getColorState(), fullUpdate);
+                updateState(HueBindingConstants.CHANNEL_BRIGHTNESS, resource.getBrightnessState(), fullUpdate);
+                updateState(HueBindingConstants.CHANNEL_SWITCH, resource.getSwitch(), fullUpdate);
+                break;
+
+            case LIGHT_LEVEL:
+                updateState(HueBindingConstants.CHANNEL_2_LIGHT_LEVEL, resource.getLightLevelState(), fullUpdate);
+                updateState(HueBindingConstants.CHANNEL_2_LIGHT_LEVEL_ENABLED, resource.getEnabledState(), fullUpdate);
+                break;
+
+            case MOTION:
+                updateState(HueBindingConstants.CHANNEL_2_MOTION, resource.getMotionState(), fullUpdate);
+                updateState(HueBindingConstants.CHANNEL_2_MOTION_ENABLED, resource.getEnabledState(), fullUpdate);
+                break;
+
+            case RELATIVE_ROTARY:
+                if (fullUpdate) {
+                    supportedChannelIds.add(HueBindingConstants.CHANNEL_2_ROTARY_STEPS);
+                }
+                updateState(HueBindingConstants.CHANNEL_2_ROTARY_STEPS, resource.getRotaryStepsState(), fullUpdate);
+                break;
+
+            case TEMPERATURE:
+                updateState(HueBindingConstants.CHANNEL_TEMPERATURE, resource.getTemperatureState(), fullUpdate);
+                updateState(HueBindingConstants.CHANNEL_2_TEMPERATURE_ENABLED, resource.getEnabledState(), fullUpdate);
+                break;
+
+            case ZIGBEE_CONNECTIVITY:
+                updateConnectivityState(resource);
+                updateState(HueBindingConstants.CHANNEL_2_ZIGBEE_STATUS, resource.getZigbeeState(), fullUpdate);
+                break;
+
+            default:
+                return false;
+        }
+        updateState(HueBindingConstants.CHANNEL_2_LAST_UPDATED, new DateTimeType(), fullUpdate);
+        return true;
+    }
+
+    /**
+     * Check the Zigbee connectivity and set the thing online status accordingly. If the thing is offline then set all
+     * its channel states to undefined, otherwise execute a refresh command to update channels to the latest current
+     * state.
+     *
+     * @param resource a Resource that potentially contains the Zigbee connectivity state.
+     */
+    private void updateConnectivityState(Resource resource) {
+        ZigbeeStatus zigbeeStatus = resource.getZigbeeStatus();
+        if (Objects.nonNull(zigbeeStatus)) {
+            logger.debug("updateConnectivityState() thingStatus:{}, zigbeeStatus:{}", thing.getStatus(), zigbeeStatus);
+            hasConnectivityIssue = zigbeeStatus != ZigbeeStatus.CONNECTED;
```

Comment:
Could this be split up?
- `CONNECTED` -> `ONLINE`
- `DISCONNECTED` -> `OFFLINE`
- `CONNECTIVITY_ISSUE` -> `OFFLINE`/`COMMUNICATION_ERROR`
- (not sure what `UNIDIRECTIONAL_INCOMING` is)

Use case: I have multiple lamps that I switch off by cutting power to them. As this is a normal operating condition, I think it makes sense to show the lamps as 'OFFLINE' instead of 'ERROR: COMM' in that state.

Reply:
Umm. No. The 'DISCONNECTED` state is never reported. If you apply/remove power to/from a device it only reports `CONNECTED` or `CONNECTIVITY_ISSUE`..

Reply:
OK, fair enough. Still, wouldn't it be better to use `OFFLINE` without error detail in that case? Rationale:
- it will very often be a normal operating condition
- the error detail (+ message) doesn't actually provide any additional information beyond 'device is not reachable'

Reply:
Ok.